src/PointwiseFunctions/GeneralRelativity/KerrSchildCoords.cpp

```diff
@@ -0,0 +1,113 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \cond
+namespace gr {
+
+KerrSchildCoords::KerrSchildCoords(const double spin_a) noexcept
+    : spin_a_(spin_a) {}
+
+void KerrSchildCoords::pup(PUP::er& p) noexcept { p | spin_a_; }
+
+template <typename DataType>
+Scalar<DataType> KerrSchildCoords::r_coord_squared(
+    const tnsr::I<DataType, 3, Frame::Inertial>& cartesian_coords) const
+    noexcept {
+  const double a_squared = square(spin_a_);
+  const DataType temp =
+      0.5 * (square(cartesian_coords.get(0)) + square(cartesian_coords.get(1)) +
+             square(cartesian_coords.get(2)) - a_squared);
```

Comment:
Isn't this `dot_product(cartesian_coords, cartesian_coords)` ?

And if there is a reason not to re-use `dot_product`, you should at least use the compile-time `get<0>(cartesian_coords)` and so on.